src/Matrix.jl

```diff
@@ -2873,6 +2873,265 @@ function rank(M::MatrixElem{T}) where {T <: FieldElement}
    return lu!(P, A)
 end
 
+@doc raw"""
+   rank_interpolation(M::MatrixElem{T}) where {T <: PolyRingElem} -> Int
+   rank_interpolation(M::MatrixElem{T}) where {T <: MPolyRingElem} -> Int
+   rank_interpolation(M::MatrixElem{T}) where {T <: AbstractAlgebra.Generic.RationalFunctionFieldElem} -> Int
+
+Returns the rank of $A$ using an interpolation-like method. 
+
+# Examples
+
+```jldoctest
+julia> Qx, x = polynomial_ring(QQ, :x);
+
+julia> rank_interpolation(matrix(Qx, 2, 2, [x 2x; x^2 1])
+2
+
+julia> Qy, y = rational_function_field(QQ, :y);
+
+julia> rank_interpolation(matrix(Qy, 2, 2, [1//y -y^2; 3 2-y])
+2
+```
+"""
+function rank_interpolation(M::MatrixElem{<: PolyRingElem{<: FieldElem}})
+   n = nrows(M)
+   m = ncols(M)
+   if is_zero(n) | is_zero(m)
+      return 0
+   end
+   Kx = base_ring(M)
+   K = base_ring(Kx)
+   #The maximum degree of det(M') is calculated where M' is an arbitrary quadratic submatrix of M.
+   min_ = min(n, m)
+   if (min_ == n)
+      maxdetdeg = sum(maximum(degree(M[i, j]) for j in 1:m) for i in 1:n)
+   else 
+      maxdetdeg = sum(maximum(degree(M[i, j]) for i in 1:n) for j in 1:m)
+   end
+   r = 0
+   eval_set = evaluation_points(K, maxdetdeg+1)
+   if !is_empty(eval_set)
+      #rank(M) is calculated by computing the rank of det_deg+1 matrices, evaluated in an element of eval_set, respectively.
+      for elem in eval_set
+         M_eval = map_entries(p -> evaluate(p, elem), M)
+         r = max(rank_interpolation(M_eval), r)
+         if (r == min_)
```

Comment:
```suggestion
         if r == min_
```
for consistency with the rest of your code